test/e2e/performanceprofile/functests/utils/tuned/tuned.go

```diff
@@ -84,38 +84,34 @@ func GetPod(ctx context.Context, node *corev1.Node) (*corev1.Pod, error) {
 }
 
 func WaitForStalldTo(run bool, interval, timeout time.Duration, node *corev1.Node) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
-		cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
-		stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
-		if err != nil {
-			return false, fmt.Errorf("failed to execute command %q on node: %q; %w", cmd, node.Name, err)
-		}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		cmd := []string{"/bin/bash", "-c", "pidof stalld"}
+		stalldPid, _ := nodes.ExecCommandOnNode(cmd, node)
 
-		_, err = strconv.Atoi(stalldPid)
+		_, err := strconv.Atoi(stalldPid)
 		if !run { // we don't want stalld to run
 			if err == nil {
-				return false, fmt.Errorf("node=%q stalld_pid=%q stalld is running when it shouldn't", node.Name, stalldPid)
+				return false, nil
 			}
+			return true, nil
 		}
 		// we want stalld to run
 		if err != nil {
-			klog.Warningf("node=%q stalld_pid=%q is not a valid pid number: %v", node.Name, stalldPid, err)
```

Comment:
Leave the warning it doesn't affect the test result and provide some good indication.

Reply:
this will spam the log and the error message was moved to a higher level for better visibility.  Knowing the pid number itself does not provide too much helpful info in this particular case.

Reply:
It would at least give us some indication about where the test get stuck and how much time it takes.